libs/store/src/lib/domain/query/query.ts

```diff
@@ -0,0 +1,243 @@
+import {
+    Type
+} from '@angular/core';
+import {
+    Observable,
+    of
+} from 'rxjs';
+import {
+    EntityCollectionService,
+    QueryParams
+} from '@ngrx/data';
+
+import { Predicate } from './grammer/predicate';
+import { QueryAdapter } from './query-adapter';
+
+export interface OrderBy <TModel, TProperty extends keyof TModel> {
+    field: TProperty;
+    order?: 'ASCENDING' | 'DESCENDING';
+}
+
+/**
+ * @todo We may need a method for end-users to add custom query parameters.
+ */
+export class Query<TModel> {
+
+    /** @hidden - stores current keyword */
+    _keyword: string;
+
+    /** @hidden - stores current predicate */
+    _predicate: Predicate<TModel>;
+
+    /** @hidden - stores current page size */
+    _pageSize: number;
+
+    /** @hidden - stores current index offset */
+    _offset: number;
+
+    /** @hidden - stores flag for suspending page reset on query change */
+    _suppressPageReset: boolean;
+
+    /** @hidden - stores curernt order bys */
+    _orderByFields: Array<OrderBy<TModel, keyof TModel>>;
+
+    /** @hidden - stores current enable count flag */
+    _includeCount: boolean;
+
+    /** @hidden - stores current selection of properties */
+    _select: string[];
```

Comment:
We should work here with typings too. 
Array<<TModel, keyof TModel> or something like this..